packages/core/src/electron-browser/menu/electron-menu-contribution.ts

```diff
@@ -152,19 +166,32 @@ export class ElectronMenuContribution implements FrontendApplicationContribution
         });
 
         registry.registerCommand(ElectronCommands.ZOOM_IN, {
-            execute: () => {
+            execute: async () => {
                 const webContents = currentWindow.webContents;
-                webContents.setZoomLevel(webContents.zoomLevel + 0.5);
+                let zoomLevel = this.roundZoomLevel(webContents.zoomLevel, ZoomLevelConstraints.VARIATION, true) + ZoomLevelConstraints.VARIATION;
+                if (zoomLevel > ZoomLevelConstraints.MAX) {
+                    zoomLevel = ZoomLevelConstraints.MAX;
+                    return;
+                };
+                this.preferenceService.set('window.zoomLevel', zoomLevel, PreferenceScope.User);
             }
         });
         registry.registerCommand(ElectronCommands.ZOOM_OUT, {
-            execute: () => {
+            execute: async () => {
                 const webContents = currentWindow.webContents;
-                webContents.setZoomLevel(webContents.zoomLevel - 0.5);
+                let zoomLevel = this.roundZoomLevel(webContents.zoomLevel, ZoomLevelConstraints.VARIATION, false) - ZoomLevelConstraints.VARIATION;
+                if (zoomLevel < ZoomLevelConstraints.MIN) {
+                    zoomLevel = ZoomLevelConstraints.MIN;
+                    return;
+                };
+                this.preferenceService.set('window.zoomLevel', zoomLevel, PreferenceScope.User);
```

Comment:
I believe we should define the `scope` as part of the schema, so that we would not need to set the scope here.